app/services/create_form_service.rb

```diff
@@ -1,7 +1,32 @@
 class CreateFormService
+  class CreateFormEvent < ApplicationRecord
+    belongs_to :user
+    belongs_to :group
+  end
+
   def create!(creator:, group:, name:)
-    form = FormRepository.create!(creator_id: creator.id, name:)
-    GroupForm.create!(group:, form_id: form.id)
+    event = begin
+      CreateFormEvent.create!(group:, form_name: name, user: creator, dedup_version: 1)
+    rescue ActiveRecord::RecordNotUnique
+      previous_event = CreateFormEvent.order(created_at: :desc).find_by(group:, form_name: name)
+      if previous_event.user == creator && previous_event.created_at > 1.second.ago
```

Comment:
Maybe it's because it's late in the day but my brain took slightly too long to figure out how this meant that the previous event was created more recently than 1 second ago, rather then more than one second ago. Reckon it's worth adding a tiny comment to help other people with slow brains?

Reply:
Yeah it's not just you, I also find the ordering in time comparisons annoying, I wish there was a more readable way of writing it. I'll add a comment, good suggestion.